### PR TITLE
Missing CtTypeReference for instanceof expressions when querying.

### PR DESCRIPTION
**Pull request by revaultch**
When rewriting a type instanceof expressions are not updated with the new type.

### CODE REVIEW

**Comment by GerardPaligot** [comment]
Thanks @revaultch and this PR should fixed this issue https://github.com/INRIA/spoon/issues/349 so that's cool! :)

File: src/test/java/spoon/test/refactoring/RefactoringTest.java [hunk 1]
```diff
@@ -89,4 +90,25 @@ public boolean matches(CtInvocation<?> element) {
...
+                final Launcher launcher = new Launcher();
+                launcher.setArgs(new String[] {
+                                "-i", "src/test/java/spoon/test/refactoring/testclasses",
+                                "-o", "target/spooned/refactoring",
+                                "-p", ThisTransformationProcessor.class.getName()
+                });
+                launcher.run();
+                final CtClass<?> aClassX = (CtClass<?>) launcher.getFactory().Type().get("spoon.test.refactoring.testclasses.AClassX");
+
+                final CtBinaryOperator<?> instanceofInvocation = aClassX.getElements(new AbstractFilter<CtBinaryOperator<?>>(CtBinaryOperator.class) {
```
Comment (GerardPaligot) [inline src/test/java/spoon/test/refactoring/RefactoringTest.java]: There is a better way to do that: use `new TypeFilter<CtBinaryOperator<?>>(CtBinaryOperator.class)`.

File: src/test/java/spoon/test/refactoring/RefactoringTest.java [hunk 2]
```diff
@@ -89,4 +90,25 @@ public boolean matches(CtInvocation<?> element) {
...
+                        @Override
+                        public boolean matches(CtBinaryOperator<?> element) {
+                                return true;
+                        }
+                }).get(0);
+                assertEquals("o instanceof spoon.test.refactoring.testclasses.AClassX", instanceofInvocation.toString());
```
Comment (GerardPaligot) [inline src/test/java/spoon/test/refactoring/RefactoringTest.java]: Can you add assertions on the left and right hand operators?

**Comment by revaultch** [comment]
I imported  SpoonFormatterPrefs.xml in Eclipse, but the resulting formatting doesn't seem to correspond to what's in the repo. Can you update said file or give instructions on the formatting?

**Comment by GerardPaligot** [comment]
Thanks for the report, I created a issue for that: https://github.com/INRIA/spoon/issues/393

**Comment by monperrus** [comment]
Thanks a lot @revaultch for this first contribution to Spoon! You can propose a PR on the README/pom.cml to add your name in the list of contributors.

Closes #349

**Pull request merged**